tests/test_autologin.py

```diff
@@ -30,39 +31,55 @@ def test_login_request():
 
 # These tests should be run last as it uses crochet, and normal scrapy spider
 # is not finalized correctly after a call to crochet.setup.
+@pytest.mark.last
+class TestAuthCookiesFromUrl(unittest.TestCase):
+    
+    url = 'http://127.0.0.1:{}/login1'.format(PORT)
+    url_no_change_cookie = 'http://127.0.0.1:{}/login2'.format(PORT)
 
+    def setUp(self):
+        self.al = AutoLogin()
+        self.mockserver = MockServer()
+        self.mockserver.__enter__()
 
-@pytest.mark.last
-def test_auth_cookies_from_url_login1():
-    al = AutoLogin()
-    url = 'http://localhost:{}/login1'.format(PORT)
-    with MockServer():
-        # no login form
+    def tearDown(self):
+        self.mockserver.__exit__(None, None, None)
+
+    def test_no_login_form(self):
         with pytest.raises(AutoLoginException) as e:
-            al.auth_cookies_from_url(url + '?hide=', 'admin', 'secret')
+            self.al.auth_cookies_from_url(
+                self.url + '?hide=', 'admin', 'secret')
         assert e.value.args[0] == 'nologinform'
-        # wrong password
+
+    def test_wrong_password(self):
         with pytest.raises(AutoLoginException) as e:
-            al.auth_cookies_from_url(url, 'admin', 'wrong')
+            self.al.auth_cookies_from_url(self.url, 'admin', 'wrong')
         assert e.value.args[0] == 'badauth'
-        # normal auth
-        cookies = al.auth_cookies_from_url(url + '?foo=', 'admin', 'secret')
+
+    def test_normal_auth(self):
+        cookies = self.al.auth_cookies_from_url(
+            self.url + '?foo=', 'admin', 'secret')
         assert {c.name: c.value for c in cookies} == {'_auth': 'yes'}
-        # redirect to the same URL
-        cookies = al.auth_cookies_from_url(url, 'admin', 'secret')
+
+    def test_redirect_to_same_url(self):
+        cookies = self.al.auth_cookies_from_url(self.url, 'admin', 'secret')
         assert {c.name: c.value for c in cookies} == {'_auth': 'yes'}
 
+    def test_proxy(self):
+        assert 'localhost' not in self.url, 'proxy_bypass bypasses localhost'
+        cookies = self.al.auth_cookies_from_url(
+            self.url, 'admin', 'secret',
+            settings={'HTTP_PROXY': 'http://127.0.0.1:8123'},
+        )
+        assert {c.name: c.value for c in cookies} == {'_auth': 'yes'}
```

Comment:
Added the check - I set a custom header and check in the ``LoginCheckProxy``, also checking that it does not work without proxy beforehand. Thank for the suggestion!